panda/src/navigation/config_navigation.cxx

```diff
@@ -0,0 +1,52 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file config_navigation.cxx
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+#include "config_navigation.h"
+
+#include "pandaSystem.h"
+#include "dconfig.h"
+#include "navMesh.h"
+#include "navMeshNode.h"
+#include "navMeshQuery.h"
+
+#if !defined(CPPPARSER) && !defined(LINK_ALL_STATIC) && !defined(BUILDING_RECASTDETOUR)
+  #error Buildsystem error: BUILDING_NAVIGATION not defined
+#endif
+
+Configure(config_navigation);
+NotifyCategoryDef(navigation, "");
+
+ConfigureFn(config_navigation) {
+  init_libnavigation();
+}
+
+ConfigVariableInt navigation_sample_config_variable
+("navigation-sample-config-variable", 25);
+
+/**
+ * Initializes the library.  This must be called at least once before any of
+ * the functions or classes in this library can be used.  Normally it will be
+ * called by the static initializers and need not be called explicitly, but
+ * special cases exist.
+ */
+void
+init_libnavigation() {
+  NavMesh::init_type();
```

Comment:
I believe initializing these types should come after the init guard (`if (initialized)`). Also, I see that config_bullet.cxx is calling the `register_with_read_factory()` member functions on various types, so I am guessing you'll need to do the same.